dagmc_stats/DagmcQuery.py

```diff
@@ -236,3 +303,268 @@ def calc_area_triangle(self, tris=[]):
+        below and update the global averages dictionary.
+
+        https://www.sciencedirect.com/science/article/pii/S0097849312001203
+
+        inputs
+        ------
+            none
+
+        outputs
+        -------
+            none
+        """
+        verts = self.get_verts()
+        self.calc_area_triangle()  # get area data if not already calculated
+        si_total = 0.  # sum of denominator
+        lri_si_total = 0.  # sum of numerator
+        for vert in verts:
+            # get adjacent triangles and their areas
+            tris = self.dagmc_file._my_moab_core.get_adjacencies(
+                vert, 2, op_type=0)
+            area_sum = self._tri_data.loc[
+                self._tri_data['tri_eh'].isin(tris)]['area'].sum()
+
+            # si = 1/3 of total area of adjacent triangles
+            si = area_sum / 3.
+            si_total += si
+
+            # calculate numerator (lri*si)
+            lri_si = float(self._vert_data.loc[
+                self._vert_data['vert_eh'] == vert]['roughness']) * si
+            lri_si_total += lri_si
+
+        # calc average according to formula 5
+        average_roughness = lri_si_total / si_total
+
+        # update global average dictionary
+        self._global_averages['roughness_ave'] = average_roughness
+
+    def calc_roughness(self):
+        """Calculate local roughness values of all the non-isolated vertices
+        and calculates the average roughness for each surface of the meshset
+        list and the average of the entire meshset list.
+
+        reference:
+        https://www.sciencedirect.com/science/article/pii/S0097849312001203
+
+        inputs
+        ------
+            none
+
+        outputs
+        -------
+            none
+        """
+        tri_vert_data, all_verts = self.__get_tri_vert_data()
+        verts = self.get_verts()
+        gc_all = self.__calc_gaussian_curvature(tri_vert_data)
+        roughness_per_vert = []
+        for vert in verts:
+            rval = self.__get_lri(vert, gc_all, tri_vert_data)
+            row_data = {'vert_eh': vert, 'roughness': rval}
+            roughness_per_vert.append(row_data)
+        self.__update_vert_data(roughness_per_vert)
+
+        # calculate average for meshset list
+        self.__calc_average_roughness()
+        
+        # calculate triangle average roughness
+        self.__calc_tri_roughness()
+        
+    def __calc_tri_roughness(self):
+        """Calculate triangle average roughness
+
+        inputs
+        ------
+            none
+
+        outputs
+        -------
+            none
+        """
+        tri_roughness = []
+        tris = self.get_tris()
+        for tri in tris:
+            three_verts = list(self.dagmc_file._my_moab_core.get_adjacencies(tri, 0, op_type=1))
+            sum_lr = self._vert_data.loc[
+                self._vert_data['vert_eh'].isin(three_verts)]['roughness'].sum()
+            rval = sum_lr/3
```

Comment:
```suggestion
            rval = sum_lr/3.0
```

Reply:
I assume this is to make it python 2 compatible?

Reply:
yes. we worked through a bug with the tests not matching expected results because of integer division and now I am being nitpicky about removing it all.